housekeeping/build.sh

```diff
@@ -50,7 +50,6 @@ else
     fi
     printf "⚡️ only tests run %s\n"
     rm -rf ~/.cargo/.package-cache
-    rm Cargo.lock
```

Comment:
Because `Cargo.lock` is used to control versions of indirect dependencies and provide consistent builds (including for CI) that are less dependant on external factors. Cargo is designed with this usage in mind: https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

With this `rm` in place the reproducibility of builds is violated, as we see in the recent PRs. Furthermore, we end up resorting to using `.toml` for this purpose, which does not correlate with intentions/goals of the build system. (in other words, we introduce костыли :c)

What do you mean by "freeze"?

Reply:
> with the updated dependencies

Exactly, it defeats the whole purpose of `.lock`, since it doesn't lock dependencies anymore.